File: Semestrul 4/MPP2/ConcursAtletism/src/main/java/ro/mpp2025/repository/IRepository.java
package ro.mpp2025.repository;

import ro.mpp2025.domain.Entity;

public interface IRepository<ID, T extends Entity<ID>> {
    T findOne(ID id);
    Iterable<T> findAll();
    T save(T entity);
    T delete(ID id);
    T update(T entity);
}

File: Semestrul 4/MPP2/Problema8_FX_Csharp/Problema8_FX_Csharp/Repository/IRepository.cs
using Problema8_FX_Csharp.Domain;

namespace Problema8_FX_Csharp.Repository;

public interface IRepository<T, ID> where T : Entity<ID>
{
    T FindOne(ID id);
    IEnumerable<T> FindAll();
    void Save(T entity);
    void Delete(ID id);
    void Update(T entity);
}